src/lamplib/src/genny/tasks/auto_tasks.py

```diff
@@ -281,17 +358,19 @@ def all_tasks(self) -> List[GeneratedTask]:
         # Double list-comprehensions always read backward to me :(
         return [task for workload in self.all_workloads() for task in workload.all_tasks()]
 
-    def variant_tasks(self, build: CurrentBuildInfo):
+    def variant_tasks(self, build: CurrentBuildInfo) -> List[GeneratedTask]:
         """
         :return: Tasks to schedule given the current variant (runtime)
         """
         return [task for workload in self.all_workloads() for task in workload.variant_tasks(build)]
 
-    def patch_tasks(self) -> List[GeneratedTask]:
+    def patch_tasks(self, build: CurrentBuildInfo) -> List[GeneratedTask]:
         """
         :return: Tasks for modified workloads current variant (runtime)
         """
-        return [task for workload in self.modified_workloads() for task in workload.all_tasks()]
+        return [
+            task for workload in self.modified_workloads() for task in workload.variant_tasks(build)
```

Comment:
patch_tasks now only schedules modified workloads if they are compatible with the variant. This is a change from before where patch_tasks would schedule all modified workloads for all variants selected. This meant if you modified a workload requiring a replica set, and selected patch_auto_tasks on the standalone variant, the task would be scheduled and proceed to fail.